### DISCUSSION TITLE
automatic selection of local network or public domain

### DISCUSSION THREAD

**Discussion by alx-xlx** [vrtmrz/obsidian-livesync#388] [Ideas]
I have couchdb running on raspberry pi in my local wifi network. I have also exposed the couchdb to the internet, but it is super slow as compared to being in the localnetwork.

I wish if there was a condition that if the pc or device is connected to a specific wifi (in which the couchdb is running) then the obsidian-livesync should use the http://couchdb.pi:8456 and not https://example.com. This was if I am at home then the livesync would be super fast.

is this somethings that's possible ?

**Comment by vrtmrz** [comment]
Thank you for your idea!

It would be nice for some environments, but we might become a bit more greedy. (e.g.; with/without VPN or something).
However, we cannot know the connection type directly. Thus, we need some more ideas for the automatic selection. 

Nevertheless, a feature of switching the remote database could be useful. I hope to keep this idea to be marked.

**Comment by SystemZ** [comment]
Consider using public domain inside LAN.

Then your LAN DNS server can override IP of server to point to local server.
When you leave home it seamlessly switch without need to reconfigure any apps.

**Reply by alx-xlx** [reply]
I am using the Cloudflared tunnel to point the paperless.example.com to 192.168.0.103:4381 (which runs the paperless).

can this be set on pihole ??

**Reply by SystemZ** [reply]
Probably yes, I personally use similar AdGuard Home for that.

Remember that it also needs same URL on LAN so you need some proxy (or config with cert for CouchDB) for TLS and run it on default port (443).

**Reply by alx-xlx** [reply]
is there some kind of techincal name for this ? or where can I learn more about this thing ? maybe a guide on how to set this thing

**Reply by SystemZ** [reply]
I hope it will be good starting point:

https://www.reddit.com/r/homelab/comments/aqe90y/i_recently_learned_what_a_reverse_proxy_is_how/
